src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHost.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
+{
+    // Somewhat similar to https://github.com/dotnet/project-system/blob/fa074d228dcff6dae9e48ce43dd4a3a5aa22e8f0/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs
+    //
+    // This class is responsible for intializing the Razor ProjectSnapshotManager
+    [Export]
+    internal class RazorProjectHost : OnceInitializedOnceDisposedAsync
+    {
+        private readonly IUnconfiguredProjectCommonServices _commonServices;
+        private readonly Workspace _workspace;
+        private readonly AsyncSemaphore _lock;
+
+        private ProjectSnapshotManagerBase _projectManager;
+        private HostProject _current;
+        private IDisposable _subscription;
+
+        [ImportingConstructor]
+        public RazorProjectHost(
+            IUnconfiguredProjectCommonServices commonServices,
+            [Import(typeof(VisualStudioWorkspace))] Workspace workspace)
```

Comment:
Hmmm, does the Roslyn project system end up doing this? Only reason why I ask is because of all the work we just did to not use the VisualStudioWorkspace 😄.

Reply:
I think this is fine. We only care about the main VS workspace in VS for windows. 

The other workspaces don't ever have real projects.